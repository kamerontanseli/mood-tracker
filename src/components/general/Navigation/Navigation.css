.Navigation {
  margin: 0;
  list-style: none;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  background-color: #fff;
  border: 1px solid #eee;
  border-radius: 4px;
  box-shadow: 0px 2px 5px -1px #eee;
  padding: 20px;
}

.Navigation-brand {
  flex-grow: 1;
  font-size: 1.5em;
  font-weight: bold;
}

.Navigation-link {
  margin-right: 20px;
}

.Navigation-link:last-child {
  margin-right: 0px;
}

.Navigation-brand > a, .Navigation-link > a {
  text-decoration: none;
  color: #333;
  padding-bottom: 5px;
}

.Navigation-link > a:hover {
  border-bottom: 1px solid #000;
}