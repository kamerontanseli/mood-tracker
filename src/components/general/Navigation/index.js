import "./Navigation.css";
import React from "react";
import { Link } from "react-router-dom";

const Navigation = () => (
  <nav>
    <ul className="Navigation">
      <li className="Navigation-brand">
        <Link to="/">Moodr</Link>
      </li>
      <li className="Navigation-link">
        <Link to="/">Insights</Link>
      </li>
      <li className="Navigation-link">
        <Link to="/survey">Add Insight</Link>
      </li>
    </ul>
  </nav>
);

export default Navigation;
