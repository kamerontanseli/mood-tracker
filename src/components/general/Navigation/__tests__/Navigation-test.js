import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from '..'

jest.mock('react-router-dom', () => ({
  Link: p => <a {...p} />
}))

describe('render', () => {
  it('should render', () => {
    const snapshot = renderer.create(
      <Navigation />
    );
    expect(snapshot).toMatchSnapshot()
  })
})