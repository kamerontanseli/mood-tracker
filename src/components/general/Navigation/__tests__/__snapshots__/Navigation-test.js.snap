// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
<nav>
  <ul
    className="Navigation"
  >
    <li
      className="Navigation-brand"
    >
      <a
        to="/"
      >
        Moodr
      </a>
    </li>
    <li
      className="Navigation-link"
    >
      <a
        to="/"
      >
        Insights
      </a>
    </li>
    <li
      className="Navigation-link"
    >
      <a
        to="/survey"
      >
        Add Insight
      </a>
    </li>
  </ul>
</nav>
`;
